src/vm/dllimport.cpp

```diff
@@ -6362,6 +6371,55 @@ NATIVE_LIBRARY_HANDLE NDirect::LoadLibraryModuleViaHost(NDirectMethodDesc * pMD,
     return (NATIVE_LIBRARY_HANDLE)hmod;
 }
 
+NATIVE_LIBRARY_HANDLE NDirect::LoadLibraryModuleViaCallBack(NDirectMethodDesc * pMD, LPCWSTR wszLibName)
+{
+#ifndef PLATFORM_UNIX
+    if (IsWindowsAPI(wszLibName))
+    {
+        // Prevent Overriding of Windows API sets.
+        return NULL;
+    }
+#endif // !PLATFORM_UNIX
+
+    DWORD dllImportSearchPathFlags = 0;
+    BOOL hasDllImportSearchPathFlags = pMD->HasDefaultDllImportSearchPathsAttribute();
+    if (hasDllImportSearchPathFlags)
+    {
+        dllImportSearchPathFlags = pMD->DefaultDllImportSearchPathsAttributeCachedValue();
+        if (pMD->DllImportSearchAssemblyDirectory())
+            dllImportSearchPathFlags |= DLLIMPORTSEARCHPATH_ASSEMBLYDIRECTORY;
+    }
+
+    GCX_COOP();
+
+    struct {
+        STRINGREF libNameRef;
+        OBJECTREF assemblyRef;
+    } protect;
+
+
+    Assembly* pAssembly = pMD->GetMethodTable()->GetAssembly();
+    protect.libNameRef = StringObject::NewString(wszLibName);
+    protect.assemblyRef = pAssembly->GetExposedObject();
```

Comment:
This has GC hole. You are assigning object references without them being protected.

This exact situation is described in "Your first GC hole" in https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-code-guide.md#2.1.2.